constraint/pkg/client/client_test.go

```diff
@@ -1039,6 +1041,261 @@ func TestClient_AddConstraint(t *testing.T) {
 	}
 }
 
+// TestClient_AddConstraint_withDefaultParams makes sure that we can inject
+// OpenAPIV3 Default parameters into a constraint that does not define its parameters
+// but its template defines defaults for the parameters in the spec's schema.
+func TestClient_AddConstraint_withDefaultParams(t *testing.T) {
+	defaults := []interface{}{"foo", "bar", "baz"}
+	defaults2 := []interface{}{"a", "b", "c"}
+	defaultsJSON := apiextensions.JSON(defaults)
+	defaults2JSON := apiextensions.JSON(defaults2)
+	defaultBuzzJSON := apiextensions.JSON("NOTbuzz")
+
+	target := &handlertest.Handler{}
+	template := cts.New()
+	constraint := cts.MakeConstraint(t, "Fakes", "fakes")
+
+	regoDriver, err := rego.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defaultHandled := map[string]bool{handlertest.TargetName: true}
+	tcs := []struct {
+		name string
+		// validationSpec will be used to mutate the CRD Spec to show different cases.
+		validationSpec *templates.Validation
+		// constraintSpec will be used to simulate the spec for a constraint
+		// like for a constraint that actually defines a parameter.
+		constraintSpec map[string]interface{}
+
+		wantSpec               map[string]interface{}
+		wantHandled            map[string]bool
+		wantAddConstraintError error
+		wantGetConstraintError error
+
+		// driver is pulled out to test actual driver implementations.
+		// defulat is fake.
+		driver drivers.Driver
+	}{
+		// this no op test case is tested implicitly in all
```

Comment:
no need for per-test comments.

Reply:
I find that comments on tests help me understand why the test is there to begin with. I'm curious if others feel the same. If you feel strongly against adding these, I can take the comments out.

Reply:
Strongly against. The name and parameters are descriptive enough, with the comments mostly duplicating the code. Comments that are tightly coupled to code also tend to go stale.

Reply:
test case comments removed.